### PR TITLE
Expose the upgrade tool in docs.daml.com

### PR DESCRIPTION
**Pull request by hrischuk-da**

### CODE REVIEW

**Review by bartfaitamas — APPROVED** [review]

File: docs/2.10.0/docs/upgrade/automation.rst [hunk 1]
```diff
@@ -13,6 +13,23 @@ specific to an individual application, just like the upgrade models.
 Nevertheless, we have found that the pattern shown here
 occurs frequently.
 
+This section describes manual upgrading steps when deploying updated Daml
+application code, by migrating contracts from one version to another.  A
+production grade, optimized migration tool that automates this application
+upgrade process is available but it does require a Daml Enterprise license.
+Its PDF manual explains how a Daml developer implements a smart contract
+upgrade migration. The manual also  advises an operations team on how to
```
Comment (cathyjung-da) [inline docs/2.10.0/docs/upgrade/automation.rst:21]:
```suggestion
upgrade migration. The manual also advises an operations team on how to
```

File: docs/2.9.4/docs/upgrade/automation.rst
Comment (cathyjung-da) [inline docs/2.9.4/docs/upgrade/automation.rst:1]: I recommend breaking down the sentence "The tool is designed to accommodate: working with..." into the following list:

The tool is designed to accommodate:
- Working with a stale view of the ledger’s state, such as after a partial migration;
- Reducing effort by generating boilerplate code for unchanged templates;
- Providing idempotent operations where possible;
- Handling very large ACS sizes; and
- Robustly handling rejected ledger commands, such as by archiving a contract in the same Daml transaction that creates an upgraded replacement, ensuring that duplicate replacement contracts are not created.

File: docs/3.1/docs/upgrade/automation.rst
Comment (cathyjung-da) [inline docs/3.1/docs/upgrade/automation.rst:1]: I recommend breaking down the sentence "The tool is designed to accommodate: working with..." into the following list:

The tool is designed to accommodate:
- Working with a stale view of the ledger’s state, such as after a partial migration;
- Reducing effort by generating boilerplate code for unchanged templates;
- Providing idempotent operations where possible;
- Handling very large ACS sizes; and
- Robustly handling rejected ledger commands, such as by archiving a contract in the same Daml transaction that creates an upgraded replacement, ensuring that duplicate replacement contracts are not created.

File: docs/2.8.9/docs/upgrade/automation.rst
Comment (cathyjung-da) [inline docs/2.8.9/docs/upgrade/automation.rst:1]: I recommend breaking down the sentence "The tool is designed to accommodate: working with..." into the following list:

The tool is designed to accommodate:
- Working with a stale view of the ledger’s state, such as after a partial migration;
- Reducing effort by generating boilerplate code for unchanged templates;
- Providing idempotent operations where possible;
- Handling very large ACS sizes; and
- Robustly handling rejected ledger commands, such as by archiving a contract in the same Daml transaction that creates an upgraded replacement, ensuring that duplicate replacement contracts are not created.

File: docs/2.10.0/docs/upgrade/automation.rst [hunk 2]
```diff
@@ -13,6 +13,23 @@ specific to an individual application, just like the upgrade models.
 Nevertheless, we have found that the pattern shown here
 occurs frequently.
 
+This section describes manual upgrading steps when deploying updated Daml
+application code, by migrating contracts from one version to another.  A
```
Comment (cathyjung-da) [inline docs/2.10.0/docs/upgrade/automation.rst:17]:
```suggestion
application code, by migrating contracts from one version to another. A
```

File: docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst [hunk 3]
```diff
@@ -81,10 +81,11 @@ are:
 
 -  The implementation of an interface instance can be changed;
 
--  A new interface instance can be added to a template.
-
-The following table summarizes the changes supported by that SCU. Consult
-the detailed documentation below for additional information.
+The following table summarizes the changes supported by SCU. Consult the
+subsequent sections below for additional information.  For application updates
```
Comment (cathyjung-da) [inline docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst:85]:
```suggestion
sections below for additional information. For application updates
```

File: docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst [hunk 4]
```diff
@@ -81,10 +81,11 @@ are:
...
+that are not covered by SCU, consult the :ref:`Automating the Upgrade Process
+<upgrade-automation>` section which describes an upgrade tool that can be used
```
Comment (cathyjung-da) [inline docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst:87]:
```suggestion
<upgrade-automation>` section, which describes an upgrade tool for
```

File: docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst [hunk 5]
```diff
@@ -81,10 +81,11 @@ are:
...
+to migrate contracts from an old version to a new version.
```
Comment (cathyjung-da) [inline docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst:88]:
```suggestion
migrating contracts from an old version to a new version.
```

File: docs/2.10.0/docs/upgrade/automation.rst
Comment (cathyjung-da) [inline docs/2.10.0/docs/upgrade/automation.rst:1]: I recommend breaking down the sentence "The tool is designed to accommodate: working with..." into the following list:

The tool is designed to accommodate:
- Working with a stale view of the ledger’s state, such as after a partial migration;
- Reducing effort by generating boilerplate code for unchanged templates;
- Providing idempotent operations where possible;
- Handling very large ACS sizes; and
- Robustly handling rejected ledger commands, such as by archiving a contract in the same Daml transaction that creates an upgraded replacement, ensuring that duplicate replacement contracts are not created.

File: docs/2.8.9/docs/upgrade/automation.rst [hunk 6]
```diff
@@ -13,6 +13,23 @@ specific to an individual application, just like the upgrade models.
 Nevertheless, we have found that the pattern shown here
 occurs frequently.
 
+This section describes manual upgrading steps when deploying updated Daml
+application code, by migrating contracts from one version to another.  A
```
Comment (cathyjung-da) [inline docs/2.8.9/docs/upgrade/automation.rst:17]:
```suggestion
application code, by migrating contracts from one version to another. A
```

File: docs/2.8.9/docs/upgrade/automation.rst [hunk 7]
```diff
@@ -13,6 +13,23 @@ specific to an individual application, just like the upgrade models.
...
+production grade, optimized migration tool that automates this application
```
Comment (cathyjung-da) [inline docs/2.8.9/docs/upgrade/automation.rst:18]:
```suggestion
production-grade, optimized migration tool that automates this application
```

File: docs/2.8.9/docs/upgrade/automation.rst [hunk 8]
```diff
@@ -13,6 +13,23 @@ specific to an individual application, just like the upgrade models.
...
+upgrade process is available but it does require a Daml Enterprise license.
```
Comment (cathyjung-da) [inline docs/2.8.9/docs/upgrade/automation.rst:19]:
```suggestion
upgrade process is available, but it requires a Daml Enterprise license.
```

File: docs/2.8.9/docs/upgrade/automation.rst [hunk 9]
```diff
@@ -13,6 +13,23 @@ specific to an individual application, just like the upgrade models.
...
+Its PDF manual explains how a Daml developer implements a smart contract
+upgrade migration. The manual also  advises an operations team on how to
```
Comment (cathyjung-da) [inline docs/2.8.9/docs/upgrade/automation.rst:21]:
```suggestion
upgrade migration. The manual also advises an operations team on how to
```

File: docs/2.8.9/docs/upgrade/automation.rst [hunk 10]
```diff
@@ -13,6 +13,23 @@ specific to an individual application, just like the upgrade models.
...
+perform the contract migration.  
+
+The tool is designed to accommodate: working with a stale view of the ledger’s
+state (e.g., a partial migration was recently performed), reduce effort by
+boilerplate code generation for unchanged templates, it has idempotent
+operations where possible, very large ACS sizes, and robustly handles rejected
+ledger commands (e.g., archiving a contract in the same Daml transaction that
+creates an upgraded replacement helps to ensure that duplicate replacement
+contracts are not created.)  Please contact your account manager or `DA
+support <support@digitalasset.com>`_ for access to the upgrade tool.
```
Comment (cathyjung-da) [inline docs/2.8.9/docs/upgrade/automation.rst:31]:
```suggestion
support <support@digitalasset.com>`_ to access the upgrade tool.
```

File: docs/2.9.4/docs/upgrade/automation.rst [hunk 11]
```diff
@@ -13,6 +13,23 @@ specific to an individual application, just like the upgrade models.
 Nevertheless, we have found that the pattern shown here
 occurs frequently.
 
+This section describes manual upgrading steps when deploying updated Daml
+application code, by migrating contracts from one version to another.  A
```
Comment (cathyjung-da) [inline docs/2.9.4/docs/upgrade/automation.rst:17]:
```suggestion
application code, by migrating contracts from one version to another. A
```

File: docs/2.9.4/docs/upgrade/automation.rst [hunk 12]
```diff
@@ -13,6 +13,23 @@ specific to an individual application, just like the upgrade models.
...
+production grade, optimized migration tool that automates this application
```
Comment (cathyjung-da) [inline docs/2.9.4/docs/upgrade/automation.rst:18]:
```suggestion
production-grade, optimized migration tool that automates this application
```

File: docs/2.9.4/docs/upgrade/automation.rst [hunk 13]
```diff
@@ -13,6 +13,23 @@ specific to an individual application, just like the upgrade models.
...
+upgrade process is available but it does require a Daml Enterprise license.
```
Comment (cathyjung-da) [inline docs/2.9.4/docs/upgrade/automation.rst:19]:
```suggestion
upgrade process is available, but it requires a Daml Enterprise license.
```

File: docs/2.9.4/docs/upgrade/automation.rst [hunk 14]
```diff
@@ -13,6 +13,23 @@ specific to an individual application, just like the upgrade models.
...
+Its PDF manual explains how a Daml developer implements a smart contract
+upgrade migration. The manual also  advises an operations team on how to
```
Comment (cathyjung-da) [inline docs/2.9.4/docs/upgrade/automation.rst:21]:
```suggestion
upgrade migration. The manual also advises an operations team on how to
```

File: docs/2.9.4/docs/upgrade/automation.rst [hunk 15]
```diff
@@ -13,6 +13,23 @@ specific to an individual application, just like the upgrade models.
...
+perform the contract migration.     
+
+The tool is designed to accommodate: working with a stale view of the ledger’s
+state (e.g., a partial migration was recently performed), reduce effort by
+boilerplate code generation for unchanged templates, it has idempotent
+operations where possible, very large ACS sizes, and robustly handles rejected
+ledger commands (e.g., archiving a contract in the same Daml transaction that
+creates an upgraded replacement helps to ensure that duplicate replacement
+contracts are not created.)  Please contact your account manager or `DA
+support <support@digitalasset.com>`_ for access to the upgrade tool.
```
Comment (cathyjung-da) [inline docs/2.9.4/docs/upgrade/automation.rst:31]:
```suggestion
support <support@digitalasset.com>`_ to access the upgrade tool.
```

File: docs/3.1/docs/upgrade/automation.rst [hunk 16]
```diff
@@ -13,6 +13,23 @@ specific to an individual application, just like the upgrade models.
 Nevertheless, we have found that the pattern shown here
 occurs frequently.
 
+This section describes manual upgrading steps when deploying updated Daml
+application code, by migrating contracts from one version to another.  A
```
Comment (cathyjung-da) [inline docs/3.1/docs/upgrade/automation.rst:17]:
```suggestion
application code, by migrating contracts from one version to another. A
```

File: docs/3.1/docs/upgrade/automation.rst [hunk 17]
```diff
@@ -13,6 +13,23 @@ specific to an individual application, just like the upgrade models.
...
+production grade, optimized migration tool that automates this application
```
Comment (cathyjung-da) [inline docs/3.1/docs/upgrade/automation.rst:18]:
```suggestion
production-grade, optimized migration tool that automates this application
```

File: docs/3.1/docs/upgrade/automation.rst [hunk 18]
```diff
@@ -13,6 +13,23 @@ specific to an individual application, just like the upgrade models.
...
+upgrade process is available but it does require a Daml Enterprise license.
```
Comment (cathyjung-da) [inline docs/3.1/docs/upgrade/automation.rst:19]:
```suggestion
upgrade process is available, but it requires a Daml Enterprise license.
```

File: docs/3.1/docs/upgrade/automation.rst [hunk 19]
```diff
@@ -13,6 +13,23 @@ specific to an individual application, just like the upgrade models.
...
+Its PDF manual explains how a Daml developer implements a smart contract
+upgrade migration. The manual also  advises an operations team on how to
```
Comment (cathyjung-da) [inline docs/3.1/docs/upgrade/automation.rst:21]:
```suggestion
upgrade migration. The manual also advises an operations team on how to
```

File: docs/3.1/docs/upgrade/automation.rst [hunk 20]
```diff
@@ -13,6 +13,23 @@ specific to an individual application, just like the upgrade models.
...
+perform the contract migration.  
+
+The tool is designed to accommodate: working with a stale view of the ledger’s
+state (e.g., a partial migration was recently performed), reduce effort by
+boilerplate code generation for unchanged templates, it has idempotent
+operations where possible, very large ACS sizes, and robustly handles rejected
+ledger commands (e.g., archiving a contract in the same Daml transaction that
+creates an upgraded replacement helps to ensure that duplicate replacement
+contracts are not created.)  Please contact your account manager or `DA
+support <support@digitalasset.com>`_ for access to the upgrade tool.
```
Comment (cathyjung-da) [inline docs/3.1/docs/upgrade/automation.rst:31]:
```suggestion
support <support@digitalasset.com>`_ to access the upgrade tool.
```

**Review by cathyjung-da — APPROVED** [review]

**Pull request merged**